spark3/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -103,6 +106,10 @@ protected Catalog buildIcebergCatalog(String name, CaseInsensitiveStringMap opti
         String warehouseLocation = options.get("warehouse");
         return new HadoopCatalog(name, conf, warehouseLocation);
 
+      case "nessie":
+        String defaultBranch = options.getOrDefault("nessie_ref", "main");
+        String nessieUrl = options.get("nessie_url");
+        return new NessieCatalog(name, conf, defaultBranch, nessieUrl);
```

Comment:
Please have a look at #1640, I'd like to standardize how we do this. I do like using `type = nessie`, so we may want to have a lookup that points to the `NessieCatalog` implementation.